daffodil-codegen-c/src/test/resources/org/apache/daffodil/codegen/c/simple.dfdl.xsd

```diff
@@ -18,45 +18,351 @@
+        <element name="simple-unsignedByte" type="si:simple-unsignedByte"/>
+        <element name="simple-unsignedInt" type="si:simple-unsignedInt"/>
+        <element name="simple-unsignedLong" type="si:simple-unsignedLong"/>
+        <element name="simple-unsignedShort" type="si:simple-unsignedShort"/>
+        <!-- All enumerations of simple types -->
+        <element name="enum-byte" type="si:enum-byte"/>
+        <element name="enum-double" type="si:enum-double"/>
+        <element name="enum-float" type="si:enum-float"/>
+        <element name="enum-hexBinary" type="si:enum-hexBinary"/>
+        <element name="enum-hexBinaryPrefixed" type="si:enum-hexBinaryPrefixed"/>
+        <element name="enum-int" type="si:enum-int"/>
+        <element name="enum-integer" type="si:enum-integer"/>
+        <element name="enum-long" type="si:enum-long"/>
+        <element name="enum-nonNegativeInteger" type="si:enum-nonNegativeInteger"/>
+        <element name="enum-short" type="si:enum-short"/>
+        <element name="enum-unsignedByte" type="si:enum-unsignedByte"/>
+        <element name="enum-unsignedInt" type="si:enum-unsignedInt"/>
+        <element name="enum-unsignedLong" type="si:enum-unsignedLong"/>
+        <element name="enum-unsignedShort" type="si:enum-unsignedShort"/>
+        <!-- All ranges of simple types -->
+        <element name="range-byte" type="si:range-byte"/>
+        <element name="range-double" type="si:range-double"/>
+        <element name="range-float" type="si:range-float"/>
+        <element name="range-int" type="si:range-int"/>
+        <element name="range-integer" type="si:range-integer"/>
+        <element name="range-long" type="si:range-long"/>
+        <element name="range-nonNegativeInteger" type="si:range-nonNegativeInteger"/>
+        <element name="range-short" type="si:range-short"/>
+        <element name="range-unsignedByte" type="si:range-unsignedByte"/>
+        <element name="range-unsignedInt" type="si:range-unsignedInt"/>
+        <element name="range-unsignedLong" type="si:range-unsignedLong"/>
+        <element name="range-unsignedShort" type="si:range-unsignedShort"/>
+      </sequence>
+    </xs:complexType>
+  </element>
+
+  <!-- Simple types without any schema facets -->
+
+  <simpleType name="simple-boolean">
+    <restriction base="xs:boolean"/>
+  </simpleType>
+  <simpleType name="simple-byte">
+    <restriction base="xs:byte"/>
+  </simpleType>
+  <simpleType name="simple-double">
+    <restriction base="xs:double"/>
+  </simpleType>
+  <simpleType name="simple-float">
+    <restriction base="xs:float"/>
+  </simpleType>
+  <simpleType name="simple-hexBinary"
+              dfdl:length="4"
+              dfdl:lengthKind="explicit"
+              dfdl:lengthUnits="bytes">
+    <restriction base="xs:hexBinary"/>
+  </simpleType>
+  <simpleType name="simple-hexBinaryPrefixed"
+              dfdl:prefixLengthType="si:prefixedCount"
+              dfdl:lengthKind="prefixed"
```

Comment:
I didn't know about this requirement until now.  It would be a good idea to implement prefix value checking, but not in this PR (going on vacation soon).